### PR TITLE
docs: update upgrade summary per feedback

### PR DESCRIPTION
**Pull request by blackchoey**
1. Add next step for users
2. Add the same part, new part and changed part between V2 and V3
3. Change `Restore your project configuration` to `How to roll back`

### CODE REVIEW

**Comment by blackchoey** [comment]
@sffamily @jidddddd please help review the content, thanks~

File: packages/fx-core/templates/core/v3Migration/migrationReport.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # Upgrade Summary
 
-Congratulations! Your project has been upgraded to work with latest Teams Toolkit. You can visit https://aka.ms/teamsfx-v5.0-guide to learn new features in Teams Toolkit. And visit https://aka.ms/teams-toolkit-5.0-upgrade to learn more about the upgrade.
+Congratulations! You can continue your development now. Your project has been upgraded to work with latest Teams Toolkit.
```
Comment (blackchoey) [inline packages/fx-core/templates/core/v3Migration/migrationReport.md:3]: Here's the next step

File: packages/fx-core/templates/core/v3Migration/migrationReport.md [hunk 2]
```diff
@@ -1,6 +1,10 @@
...
+
+Most of existing features behave similar with previous Teams Toolkit. You can visit https://aka.ms/teamsfx-v5.0-guide to learn the new features. And read [Important changes to your development flow](#important-changes-to-your-development-flow) to understand the changes to your development flow.
```
Comment (blackchoey) [inline packages/fx-core/templates/core/v3Migration/migrationReport.md:5]: Here's the same part, new part and changed part.

File: packages/fx-core/templates/core/v3Migration/migrationReport.md (hunk 1 above)
Comment (sffamily) [inline packages/fx-core/templates/core/v3Migration/migrationReport.md:3]: latest -> the latest
as Chaoyi also recommended, move the "what's next" up, to be right after this first paragraph. if the upgrade wiki has details on what's next after upgrade.

File: packages/fx-core/templates/core/v3Migration/migrationReport.md [hunk 3]
```diff
@@ -16,7 +20,9 @@ For more detailed changes, please refer to this [wiki](https://aka.ms/teams-tool
 
 ## Important changes to your development flow
 
-1. All the environment files will be gitignored by default, you need to sync their content manually. [Learn More](https://aka.ms/teams-toolkit-5.0-upgrade#environment-management)
+There's no immediate action required from you. This part illustrates what's changed to your development flow.
+
+1. You need to sync `.env/.env.{env}` files between different machine manually. All the these files will be gitignored by default. [Learn More](https://aka.ms/teams-toolkit-5.0-upgrade#environment-management)
```
Comment (sffamily) [inline packages/fx-core/templates/core/v3Migration/migrationReport.md:25]: machine -> machines
the these -> these

Comment (blackchoey) [reply packages/fx-core/templates/core/v3Migration/migrationReport.md]: Thanks, updated

File: packages/fx-core/templates/core/v3Migration/migrationReport.md (hunk 1 above)
Comment (blackchoey) [reply packages/fx-core/templates/core/v3Migration/migrationReport.md]: Updated the wording.
Actually the sentence `You can continue your development now.` is the "what's next". Users do not need to do anything immediately after upgrade.

**Review by sffamily — APPROVED** [review]

**Comment by xzf0587** [comment]
templates/appPackage/aad.template.json is an optional file for project. 
Add an optional description in the changes for it.

**Comment by blackchoey** [comment]
> templates/appPackage/aad.template.json is an optional file for project. Add an optional description in the changes for it.

@xzf0587 thanks, will address in another PR

**Pull request merged**